src/hotspot/share/compiler/disassembler.cpp

```diff
@@ -755,6 +755,17 @@ address decode_env::decode_instructions(address start, address end, address orig
 // Each method will create a decode_env before decoding.
 // You can call the decode_env methods directly if you already have one.
 
+void* Disassembler::dll_load(char* buf, int offset, size_t buflen, char* ebuf, size_t ebuflen, outputStream* st) {
+  if (offset + strlen(hsdis_library_name) + strlen(os::dll_file_extension()) < buflen) {
+    strcpy(&buf[offset], hsdis_library_name);
+    strcat(&buf[offset], os::dll_file_extension());
```

Comment:
I think we can more simplify these code as following:

```
int n_written = snprintf(&buf[offset], buflen, "%s%s", hsdis_library_name, os::dll_file_extension();
if (n_written < buflen) {
```

`snprintf()` returns _`size`_ or more bytes if the buffer is not enough.
https://man7.org/linux/man-pages/man3/printf.3.html